src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunctionSet.cs

```diff
@@ -106,14 +134,13 @@ internal TexlFunctionSet(IEnumerable<TexlFunctionSet> functionSets)
                 throw new ArgumentNullException(nameof(functionSets));
             }
 
-            foreach (var functionSet in functionSets)
-            {
-                functionSet._guard.VerifyNoWriters();
+            List<TexlFunctionSet> tfsList = functionSets.SelectMany(fs => fs.Sets).Where(fs => fs._count > 0).ToList();
 
-                if (functionSet._count > 0)
-                {
-                    Add(functionSet);
-                }
+            // keep local (empty) set for additional functions that could be added later
+            if (tfsList.Any())
+            {
+                _composed = true;
+                _texlFunctionSets = tfsList;
```

Comment:
>        _texlFunctionSets = tfsList; [](http://example.com/codeflow?start=8&length=36)

Before - we took an (expensive) snapshot of the functionset.  So if the function set was modified later, we did not pick up those changes here. 

Now, we avoid the copy (win!) - but could that change the semantics?
Ie, do we need to verify that the incoming functionSets are not mutable?

Reply:
Here's a test demonstrating the change in behavior: #2478